helix-term/src/ui/lsp.rs

```diff
@@ -53,10 +55,42 @@ impl SignatureHelp {
 }
 
 impl Component for SignatureHelp {
+    fn handle_event(&mut self, event: &Event, _cx: &mut Context) -> EventResult {
+        let Event::Key(event) = event else {
+            return EventResult::Ignored(None);
+        };
+
+        if self.signatures.len() <= 1 {
+            return EventResult::Ignored(None);
+        }
+
+        match event {
+            alt!('p') => {
+                self.active_signature = if self.active_signature == 0 {
+                    self.signatures.len() - 1
+                } else {
+                    self.active_signature - 1
+                };
+                EventResult::Consumed(None)
+            }
+            alt!('n') => {
+                self.active_signature = if self.active_signature >= self.signatures.len() - 1 {
+                    0
+                } else {
+                    self.active_signature + 1
+                };
```

Comment:
minor style nit: these can be written a little more compactly, something like

```rust
self.active_signature = self
    .active_signature
    .checked_sub(1)
    .unwrap_or(self.signatures.len() - 1);
```

and

```rust
self.active_signature = (self.active_signature + 1) % self.signatures.len();
```

Reply:
done, thanks for the improvement